test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -490,38 +467,118 @@ public void CanWrite1DDataOfMixedTypesToExcelWorksheet()
+                dataList0.Add(data.Data);
+            }
+            Assert.AreEqual(new object[] { 1.00, 2.00, 3.00, null, null, null, null }, dataList0);
+
             rowList = list[1].GetElements();
-            Assert.AreEqual(0, rowList[0].Data);
+            List<object> dataList1 = new List<object>();
+            foreach (MirrorData data in rowList)
+            {
+                dataList1.Add(data.Data);
+            }
+            Assert.AreEqual(new object[] { 1.00, 2.00, 3.00, 4.00, 5.00, null, null }, dataList1);
 
             Assert.IsTrue(list[2].IsCollection);
             rowList = list[2].GetElements();
-            Assert.AreEqual(21029, rowList[0].Data);
+            List<object> dataList2 = new List<object>();
+            foreach (MirrorData data in rowList)
+            {
+                dataList2.Add(data.Data);
+            }
+            Assert.AreEqual(new object[] { 1.00, 2.00, 3.00, 4.00, 5.00, 6.00, 7.00 }, dataList2);
 
             Assert.IsTrue(list[3].IsCollection);
             rowList = list[3].GetElements();
-            Assert.IsNull(rowList[0].Data);
+            List<object> dataList3 = new List<object>();
+            foreach (MirrorData data in rowList)
+            {
+                dataList3.Add(data.Data);
+            }
+            Assert.AreEqual(new object[] { 1.00, 2.00, 3.00, 4.00, 5.00, null, null }, dataList3);
 
-            Assert.IsTrue(list[4].IsCollection);
-            rowList = list[4].GetElements();
-            Assert.AreEqual(-90, rowList[0].Data);
+        }
+        [Test]
+        public void Excel_OverwriteWithnull_7213()
+        {
+            // Overwrite the excel sheet with null values and it must work ok.
+            string openPath = Path.Combine(TestDirectory, @"core\excel\Excel_MAGN7213.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+            Assert.AreEqual(6, ViewModel.CurrentSpace.Nodes.Count);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+            ViewModel.HomeSpace.Run();
 
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+
+            var data = new object[] { new object[] { 2 }, new object[] { null }, new object[] {3 } };
+            AssertPreviewValue(watch.GUID.ToString(), data);
+            
         }
 
         [Test]
+        public void Excel_OverWriteEmptyList_MAGN7213()
+        {
+
+            //Overwrite an existing file with EmptyList- it must run ok
+            string openPath = Path.Combine(TestDirectory, @"core\excel\Excel_MAGN7213_2.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+            Assert.AreEqual(8, ViewModel.CurrentSpace.Nodes.Count);
```

Comment:
There are 7 nodes and not 8 in this file because of which this test is failing. Please update the test case.